esphome/components/alarm_control_panel/alarm_control_panel.cpp

```diff
@@ -34,6 +36,9 @@ void AlarmControlPanel::publish_state(AlarmControlPanelState state) {
              LOG_STR_ARG(alarm_control_panel_state_to_string(prev_state)));
     this->current_state_ = state;
     this->state_callback_.call();
+#if defined(USE_ALARM_CONTROL_PANEL) && defined(USE_CONTROLLER_REGISTRY)
```

Comment:
`USE_ALARM_CONTROL_PANEL` is pretty much guaranteed if this file is being compiled. Same for all of the rest

```suggestion
#ifdef USE_CONTROLLER_REGISTRY
```

Reply:
I have a section above that explains Why #ifdef Guards in Entity Files? The problem is they get compiled in if they get included with no entities but the `USE_ALARM_CONTROL_PANEL` isn't set if there are 0 entities.  Thats a separate issue and whatever is loading `alarm_control_panel` with no entities should probably be fixed but outside the scope of this PR